### DISCUSSION TITLE
❓ Incorrect constraint on chunk length

### DISCUSSION THREAD

**Discussion by SolomidHero** [snakers4/silero-vad#259] [Q&A] [answered]
In onnx version, there is constraint: 
```
if sr / x.shape[1] > 31.25:
```

So x.shape / sr < 0.032, i.e chunk time is less than 32 ms. But you trained on chunks of size 30 ms. It seems as bug for me, because I can't use same setup as was on training

**Comment by adamnsandle** [comment, marked as answer]
Actually VAD was trained on 16k sr using 512 chunk length and on 8k sr using 256. That's exactly 31.25 ms, but for simplicity we call it 30ms.

**Reply by SolomidHero** [reply]
> That's exactly 31.25 ms

No) That's exactly 32ms

Anyway, I got your point.
